### PR TITLE
Feature/dev 1916 api endpoint to validate network

### PR DESCRIPTION
**Pull request by croomes**
Requires https://github.com/storageos/go-api/pull/49 merged first, then importing.

This adds:
```bash
$ storageos cluster connectivity 
SOURCE              NAME                         ADDRESS             LATENCY             STATUS              MESSAGE
storageos-2-simon   storageos-1-simon.etcd       172.28.128.4:5707   11.01607ms          OK                  
storageos-2-simon   storageos-3-simon.nats       172.28.128.6:5708   11.913142ms         OK                  
storageos-2-simon   storageos-3-simon.etcd       172.28.128.6:5707   11.865232ms         OK                  
storageos-2-simon   storageos-1-simon.directfs   172.28.128.4:5703   11.76831ms          OK                  
storageos-2-simon   storageos-3-simon.directfs   172.28.128.6:5703   11.687556ms         OK                  
...
```
and `storageos node connectivity [NODES]`

It also has a summary of all results:
```bash
$ storageos cluster connectivity --format summary
OK
```
Not sure I like the `connectivity` name.  Anyone prefer `netdiag` or something else?

### CODE REVIEW

File: cli/command/formatter/connectivity.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import (
+	"fmt"
+
+	"github.com/storageos/go-api/types"
+)
+
+const (
+	connectivityTableFormat   = "table {{.Source}}\t{{.Name}}\t{{.Address}}\t{{.Latency}}\t{{.Status}}\t{{.Error}}"
+	connectivityRawFormat     = `source: {{.Source}}\nname: {{.Name}}\naddress: {{.Address}}\nlatency: {{.Latency}}\nstatus: {{.Status}}\nmessage: {{.Error}}\n`
+	connectivityQuietFormat   = "{{.Name}}\t{{.Status}}"
```
Comment (domodwyer) [inline cli/command/formatter/connectivity.go]: Seems like it isn't referenced in this PR?

Comment (domodwyer) [inline cli/command/formatter/connectivity.go]: Seems like `connectivityQuietFormat` isn't referenced in this PR?

File: cli/command/cluster/connectivity.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		Short: "Display connectivity diagnostics for the cluster",
+		Args:  cli.RequiresMinArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runConnectivity(storageosCli, opt)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVarP(&opt.quiet, "quiet", "q", false, "Only display node names")
+	flags.StringVarP(&opt.format, "format", "f", "table", "Format the output using the given Go template")
```
Comment (domodwyer) [inline cli/command/cluster/connectivity.go]: Maybe list the available options in the description?

File: cli/command/cluster/connectivity.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		Use:   "connectivity [OPTIONS]",
+		Short: "Display connectivity diagnostics for the cluster",
+		Args:  cli.RequiresMinArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runConnectivity(storageosCli, opt)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVarP(&opt.quiet, "quiet", "q", false, "Only display node names")
```
Comment (domodwyer) [inline cli/command/cluster/connectivity.go]: The description is a little off? I was expecting it to just dump node names, but it prints the status of each node too.

**Comment by croomes** [comment]
Thanks @domodwyer - should all be sorted

**Pull request merged**